### PR TITLE
Add promlog package.

### PR DESCRIPTION
**Pull request by Gouthamve**
@brian-brazil @fabxc Will open the PR against prometheus once this is merged.

---
This change is [Reviewable](https://reviewable.io/reviews/prometheus/common/99)

### CODE REVIEW

File: promlog/flag/flag.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+// LevelFlagName is the canonical flag name to configure the allowed log level
+// within Prometheus projects.
+const LevelFlagName = "log.level"
+
+// LevelFlagHelp is the help description for the log.level flag.
+const LevelFlagHelp = "Only log messages with the given severity or above. One of: [debug, info, warn, error]"
+
+// AddFlags adds the flags used by this package to the Kingpin application.
+// To use the default Kingpin application, call AddFlags(kingpin.CommandLine)
+func AddFlags(a *kingpin.Application, logLevel *promlog.AllowedLevel) {
```
Comment (brian-brazil) [inline promlog/flag/flag.go]: This couples this library to kingpin, which I understand Fabian doesn't want. This likely need to be two packages to handle this.

Comment (Gouthamve) [reply promlog/flag/flag.go]: This is in a different package :) When you vendor `prometheus/common/promlog`, `prometheus/common/promlog/flag` is not vendored.

**Comment by brian-brazil** [comment]
:+1:

**Comment by Gouthamve** [comment]
@brian-brazil Lets get this in then? I will open a PR soon for dev-2.0 (reminded by having to stash them :smile:).

**Comment by brian-brazil** [comment]
I already :+1:ed

**Pull request merged**